src/Datadog.NativeLoader/proxy.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "string.h"
+#include <corhlpr.h>
+#include <corprof.h>
+#include <vector>
+#include <functional>
+
+namespace datadog
+{
+namespace nativeloader
+{
+    const IID IID_IUnknown = {0x00000000, 0x0000, 0x0000, {0xC0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46}};
+
+    typedef HRESULT(STDMETHODCALLTYPE* dllGetClassObjectPtr)(REFCLSID, REFIID, LPVOID*);
+    typedef HRESULT(STDMETHODCALLTYPE* dllCanUnloadNow)();
+
+    class DynamicInstance
+    {
+    private:
+        std::string m_filepath;
+        IID m_clsid = IID_IUnknown;
+        bool m_loaded;
+        void* m_instance;
+        dllGetClassObjectPtr m_getClassObjectPtr;
+        dllCanUnloadNow m_canUnloadNow;
+        IClassFactory* m_classFactory;
+        ICorProfilerCallback10* m_corProfilerCallback;
+
+        void EnsureDynamicLibrary();
+        HRESULT DllGetClassObject(REFIID riid, LPVOID* ppv);
+
+    public:
+        DynamicInstance(std::string filePath, REFCLSID clsid);
+        HRESULT LoadClassFactory(REFIID riid);
+        HRESULT LoadInstance(IUnknown* pUnkOuter, REFIID riid);
+        HRESULT STDMETHODCALLTYPE DllCanUnloadNow();
+        ICorProfilerCallback10* GetProfilerCallback();
+        std::string GetFilePath();
+    };
+
+    class DynamicDispatcher
+    {
+    private:
+        std::vector<DynamicInstance*> m_instances;
+
+    public:
+        DynamicDispatcher();
+        void Add(DynamicInstance* instance);
```

Comment:
you could make it :
`void Add(std::unique_ptr<DynamicInstance>&& instance);` (and use std::make_unique when added)
In that case, you do not need to care about the destruction of those instances

Reply:
Added the change in ca10cf3, can you take a look?

Reply:
sounds good to me